.github/workflows/publish.yml

```diff
@@ -103,3 +102,14 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_CONFIG_PROVENANCE: true
+
+      - name: snapshot release
+        if: ${{ steps.changesets.outputs.published == 'false' }}
+        run: |
```

Comment:
This is an output of the above changesets action.

So the thought process here is 

-if we dont release, we can release a snapshot

- so we checkout the `develop` branch (currently)
- then run `changeset version` passing `snapshot` we can name a snapshot by passing the name to it (in this case i called it snapshot).

- then call `pnpm publish -r ...` -r causes pnpm to create the topological graph of the packages so it should be in order

- tag is passed and called snapshot for sanity and so we arent overwriting latest in any registry
- no-git-checks is a pnpm feature so that if we have git files uncommitted it will commit.

Questions I am unsure of with this flow:

If there is no changeset, i'd imagine it wouldn't publish. I say this because if i run the following locally i would get this

```
pnpm changeset version --snapshot snapshot                                                                                                                                                                                                                                                             ✔  7749  16:31:50

> forgerock-javascript-sdk@ changeset /Users/ryan.basmajian/programming/forgerock-javascript-sdk
> changeset "version" "--snapshot" "snapshot"

🦋  warn No unreleased changesets found, exiting.
```

So it exits, but i'm not really sure about the next step. it feels like maybe we would want a more sophisticated script written in JS that says if we get a process exit here or if we get that text output, then we just exist the script.

Thoughts?